src/main/scala/firrtl/annotations/MemoryInitAnnotation.scala

```diff
@@ -52,3 +52,8 @@ case class MemoryFileInlineAnnotation(
   override def initValue:    MemoryInitValue = MemoryFileInlineInit(filename, hexOrBinary)
   override def isRandomInit: Boolean = false
 }
+
+/** Defines the memory initialization target, inside or outside the `ifndef SYNTHESIS` block */
+trait MemoryInitTargetAnnotation extends NoTargetAnnotation
+case object MemoryNoSynthInit extends MemoryInitTargetAnnotation
+case object MemorySynthInit extends MemoryInitTargetAnnotation
```

Comment:
```suggestion
sealed trait MemoryInitTargetAnnotation extends NoTargetAnnotation
case object MemoryNoSynthInit extends MemoryInitTargetAnnotation
case object MemorySynthInit extends MemoryInitTargetAnnotation
```
You don't use `MemoryInitTargetAnnotation` for anything, is it really needed?

Reply:
Really not needed. I was doing a `match case` before.. but not anymore.
Removed.